### DISCUSSION TITLE
Masks instead of bounding boxes

### DISCUSSION THREAD

**Discussion by thepaulchee** [ultralytics/yolov5#6747] [Q&A] [answered]
Hi,

I'm new to YOLOv5 and object detection in general. Is there an option for YOLOv5 to output masks instead of bounding boxes for classification?

If not, can you advise on what is the proper workflow following training using YOLOv5 to get those masks? Specifically, I would like to get the surface area of the object detected.

Thanks!

**Comment by glenn-jocher** [comment, marked as answer]
@thepaulchee hi thanks for asking about masks! Segmentation (masks) is in our product pipeline for later this year. Currently YOLOv5 supports classification (with classification branch) and detection.

<img width="2163" alt="Screenshot 2021-11-22 at 18 26 43" src="https://user-images.githubusercontent.com/26833433/155314846-96ae3924-f580-4298-9c9c-18454b982795.png">

**Comment by thepaulchee** [comment]
Thanks! Good to know it's in the pipeline! 👍🏻

**Comment by kpbhat25** [comment]
Is this pipeline done ?
Needed masks instead of bounding box!
Thank you
